index.js

```diff
@@ -194,6 +215,51 @@ FSWatcher.prototype._throttle = function(action, path, timeout) {
   return throttled[path];
 };
 
+// Private method: Awaits write operation to finish
+//
+// * path    - string, path being acted upon
+// * threshold - int, time in milliseconds a file size must be fixed before acknowledgeing write operation is finished
+// * callback - function, callback to call when write operation is finished 
+// Polls a newly created file for size variations. When files size does not change for 'threshold'
+// milliseconds calls callback.
+FSWatcher.prototype._awaitWriteFinish = function(path, threshold, callback){
+  var timeoutHandler;
+
+  var awaitWriteFinish = function(prevStat){
+    fs.stat(path, function(err, curStat){
+      if(err){
+        // if the file have been erased, the file entry in _pendingWrites will
+        // be deleted in the unlink event.
+        if(err.code == 'ENOENT') return;
+
+        return callback(err);
+      } 
+        
+      var now = new Date();
+      if(this._pendingWrites[path] === undefined){
+        this._pendingWrites[path] = {
+          creationTime: now,
+          cancelWait: function(){
+            delete this._pendingWrites[path];
+            clearTimeout(timeoutHandler);
+            return callback();
+          }.bind(this)
+        }
+        return timeoutHandler = setTimeout(awaitWriteFinish.bind(this, curStat), this.options.interval);
```

Comment:
Conflating the use of the `interval` option here, which is intended (and documented) as applying to the `usePolling` option. This probably needs a dedicated property for `awaitWriteFinishInterval`.